python/plugins/processing/algs/gdal/gdalcalc.py

```diff
@@ -131,38 +206,35 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         #if debug:
         #    arguments.append('--debug')
         arguments.append('-A')
-        arguments.append(self.getParameterValue(self.INPUT_A))
-        if self.getParameterValue(self.BAND_A):
-            arguments.append('--A_band ' + self.getParameterValue(self.BAND_A))
-        if self.getParameterValue(self.INPUT_B):
+        arguments.append(self.parameterAsLayer(parameters, self.INPUT_A, context).source())
+        if self.parameterAsString(parameters, self.BAND_A, context):
+            arguments.append('--A_band ' + self.parameterAsString(parameters, self.BAND_A, context))
+        if self.parameterAsLayer(parameters, self.INPUT_B, context):
```

Comment:
For the optional layers you would go:

    if self.INPUT_B in parameters and parameters[self.INPUT_B] is not None:
        layer_b = self.parameterAsRasterLayer(parameters, self.INPUT_B, context)
        if layer_b is None:
             raise QgsProcessingException(self.invalidRasterError(parameters, self.INPUT_B))
        arguments.append(layer_b.source())